core/src/test/java/com/basistech/ninja/NetworkTest.java

```diff
@@ -176,25 +155,37 @@ public void testFullthreeLayerDeltas() {
                 1, 2, 3, 4, 5,
                 -6, -7, -8, -9, -10
         );
-        //Network net = new Network(w1, w2);
 
         SimpleMatrix x = new SimpleMatrix(3, 1, true, 0, 0, 0);
         SimpleMatrix y = new SimpleMatrix(2, 1, true, 0, 0);
 
         // feedforward to compute z and a for each layer
-        SimpleMatrix a1 = addBiasUnit(x);
+        SimpleMatrix a1 = Network.addBiasUnit(x);
         SimpleMatrix z2 = w1.mult(a1);
-        SimpleMatrix a2 = addBiasUnit(Functions.apply(Functions.SIGMOID, z2));
+        SimpleMatrix a2 = Network.addBiasUnit(Functions.apply(Functions.SIGMOID, z2));
         SimpleMatrix z3 = w2.mult(a2);
         SimpleMatrix a3 = Functions.apply(Functions.SIGMOID, z3);
 
         // backprop to compute d for each layer (one d for every a)
         SimpleMatrix d3 = a3.minus(y);
-        SimpleMatrix d2 = w2.transpose().mult(d3).elementMult(addBiasUnit(z2));
-        d2 = stripBiasUnit(d2);
+        SimpleMatrix d2 = w2.transpose().mult(d3).elementMult(Functions.apply(
+                Functions.SIGMOID_PRIME, Network.addBiasUnit(z2)));
+        //d2 = Network.stripBiasUnit(d2);
```

Comment:
Looks good, but I think we *do* need to strip the bias here, and
in the `Network` code (there a TODO).  Also, I think we should
be testing against the raw values we actually computed (e.g.
0.393) and remove/comment the manual test computations.
I'll merge this to master, but leave this branch and push suggested
changes here.